### ISSUE TITLE
Allow featured collections installation to be specified on a per repository basis

### ISSUE THREAD

**Issue by kyleschlosser** [kabanero-io/kabanero-operator#63]
Currently whether to install the featured collections is on a per Kabanero instance basis, but this should ideally be modeled as per repository. Featured collections are to be installed by default -- if the configuration value is unspecified, then featured collection installation is enabled. 

The existing repository modeling is: 
```
repositories: 
- name: myrepo
  url: http://....
```

Proposed: 
```
repositories: 
- name: myrepo
  url: http://....
  installFeaturedCollections: true
```

**Comment by kaczyns** [comment]
After talking with @stephenkinder today, it sounds like "featuredCollections" refers to the default collections provided by Kabanero.  Any additional collections would be provided by URLs in the `repositories` section.  There is no instance of a "featured" collection in these repositories.

The current behavior should be that if "featuredCollections" are enabled, then the default Kabanero collections are installed.  Then, any repositories are processed and all collections in those repositories are installed.

@kyleschlosser please let us know you have a different thought.  It sounds like `installFeaturedCollections` on a specific repository is not necessary.

**Comment by kaczyns** [comment]
@tspewak and I spoke with @kyleschlosser today.  My understanding of what a featured collection is, is incorrect.  It should be possible to mark a collection within an index (repository) as 'featured'.  When a `Kabanero` instance is created, the spec should indicate whether or not featured collections are automatically activated (featuredCollections=true/false).  If they are automatically activated, then a `Collection` object is created for each collection in the repository that is marked 'featured'.  If they are not automatically activated, then no `Collection` objects are created, and the individual collections must be activated manually, either by creating a `Collection` object using `kubectl` or `oc`, or by using proposed CLI tools which are still under development.

The current state of the kabanero_controller is correct - in that if "featuredCollections" is true, then all of the collections in the default index are created.  This is because the controller makes an assumption that all of the collections are featured.  This assumption was not communicated well.

This issue should move the `enableFeatured` attribute of the `Kabanero` Spec.Collections section, into the Spec.Collections.Repositories section, so that `enableFeatured` can be enabled per-repository.  We should validate that the `Kabanero` controller correctly processes multiple repositories where `enableFeatured` is enabled for one, but not for the other.

For each repository, if `enableFeatured` is `false` then no `Collection` objects should be created for that repository.  If `enableFeatured` is `true`, then we should read a field called `featuredProjects` in the `index.yaml` for the repository, and we should create `Collection` objects for each collection name that matches a project in `featuredProjects`.  See [this commit](https://github.com/kabanero-io/kabanero-collection/commit/851569121187a6dd6bb278e8e287f1c1469a261c) to see where the `featuredProjects` was added to the index.

**Comment by stephenkinder** [comment]
There are a couple of issues I see here:

Kabanero uses the term "featured collection" on the website to name the community-curated collections.  Using the term in the Kabanero website and having an alternate terminology in the Kabanero CRD is confusing.

I'm not 100% sure what value the switch provides, we had the notion of a Collection has been activated. If this is the equivalent function, perhaps a better name might be `activateCollections=automatic` or some variant.

The Kabanero CRD represents a logical group of collections that are managed as a Kabanero Instance.  That said, we don't document the behavior when the list of collections contain conflicting collection definitions.

**Comment by kaczyns** [comment]
@kyleschlosser there is still some confusion about what a featured collection is.  Please see @stephenkinder's comments above.  I am assuming that your interpretation of a featured collection is coming from https://github.com/kabanero-io/kabanero-foundation/issues/2.

When I read that, it's a little ambiguous to me.  Perhaps there is some additional information that I'm not privy to yet.  It's not clear from that issue, whether a featured collection is a subset of an arbitrary collection, or the name of the community-curated collections.

**Comment by kaczyns** [comment]
Going to speak with @nateziemann about this when he's back from vacation, as that's where the requirement for auto-creating a subset of collections in a repository came from.  We want to be sure that the name we pick is appropriate.  There is concern that the name `featuredCollection` is now overloaded.

**Comment by kaczyns** [comment]
I am moving this out of epic #66 until we have a resolution on what `featuredCollection` really means.

**Comment by nateziemann** [comment]
I believe we cleared this up on Friday. So this can get targeted for an iteration now.

**Comment by kaczyns** [comment]
Yes Nate and I spoke about this on Friday.  We should be referring to the collections that we automatically activate as `default` as this is a technical / industry standard term.  The new modeling should be:
```yaml
repositories: 
- name: myrepo
  url: http://....
  activateDefaultCollections: true
```
You can remove the existing `enableFeatured` attribute on the `collections` configuration.

As we discussed, the default value will be `false` because that's how Go works, and that's OK.  If the customer wants us to enable the collections they will have to set it to 'true'.  Our examples should be changed to do this (set to true for any repositories we define).

You will need to coordinate this with @kyleschlosser's changes for the v1beta1 modeling.  I am not sure which will go first.
